src/Services/ArrayInterface.php

```diff
@@ -6,32 +6,194 @@
 interface ArrayInterface
 {
     /**
-     * to check whether the passed value is an Array or not.
+     * method returns true if the given value in an array and false it not
      *
      * @param $value
      * @return bool
      */
     public function isArray($value): bool;
 
     /**
+     * method returns true if the given array is an associative array and false if not
+     *
+     * @param array $arr
+     * @return bool
+     */
+    public function isAssoc(array $arr): bool;
+
+    /**
+     * method flattens the multi dimensional array into a single dimensional array
+     *
      * @param array $arr
-     * @param mixed ...$values
+     * @return void
+     */
+    public function flatten(array &$arr): void;
+
+    /**
+     * method inserts the specified value at the specified position
+     *
+     * @param array $arr
+     * @param int $position
+     * @param $value
+     * @return void
+     */
+    public function insert(array &$arr, int $position, $value): void;
+
+    /**
+     * method returns range of items in an array by using the "slicing :" operator
+     *
+     * @param array $arr
+     * @param string $range // "1:", "1:4"
      * @return array
      */
-    public function add(array $arr, ...$values): array;
+    public function slice(array $arr, string $range): array;
 
     /**
+     * method will push elements onto the end of an array
+     *
      * @param array $arr
-     * @param string $key
-     * @return mixed
+     * @param mixed ...$elements
+     * @return array
      */
-    public function get(array $arr, string $key);
+    public function append(array $arr, ...$elements): array;
 
     /**
+     * method will push elements onto the beginning of an array
+     *
+     * @param array $arr
+     * @param mixed ...$elements
+     * @return array
+     */
+    public function prepend(array $arr, ...$elements): array;
+
+    /**
+     * method returns two arrays, one containing the keys, and the other containing the values of the given array
+     *
+     * @param array $arr
+     * @return array
+     */
+    public function divide(array $arr): array;
```

Comment:
I wish we could create better name for this method.

Reply:
return [array_keys($arr), array_values($arr)] --- enhancement